subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/FingerprintCompareStrategy.java

```diff
@@ -29,9 +30,16 @@
  */
 public interface FingerprintCompareStrategy {
     /**
-     * @see FileCollectionFingerprint#visitChangesSince(FileCollectionFingerprint, String, boolean, ChangeVisitor)
+     * Visits the changes to file contents since the given fingerprint, subject to the given filters.
+     *
+     * @return Whether the {@link ChangeVisitor} is looking for further changes. See {@link ChangeVisitor#visitChange(Change)}.
      */
     boolean visitChangesSince(ChangeVisitor visitor, Map<String, FileSystemLocationFingerprint> current, Map<String, FileSystemLocationFingerprint> previous, String propertyTitle, boolean includeAdded);
```

Comment:
Can we move this method out of here? All it does is it iterates the maps, no?

Reply:
I removed the method from the interface.